src/Stratis.Bitcoin.IntegrationTests/RPC/RPCTestsMutable.cs

```diff
@@ -21,11 +21,18 @@ public void TestRpcGetBalanceIsSuccessful()
                 builder.StartAll();
                 RPCClient rpcClient = node.CreateRPCClient();
 
-                TestHelper.MineBlocks(node, 2);
-                TestHelper.WaitLoop(() => node.FullNode.GetBlockStoreTip().Height == 2);
+                int maturity = (int)KnownNetworks.RegTest.Consensus.CoinbaseMaturity;
+
+                // This shouldn't be necessary but is required to make this test pass.
+                // - looks like a bug in the wallet?
+                maturity--; 
```

Comment:
Hmm, you may be onto something there, although I need to investigate in further detail:

`if (coin.IsCoinBase() && nSpendHeight - coin.nHeight < COINBASE_MATURITY)`

One way to know for sure would be to connect an SBFN regtest to a bitcoind regtest and try to create such a transaction as early as possible on bitcoind to see if SBFN accepts it.

Reply:
Ok, so if we think about this:

1. Genesis has height = 0
2. Let's say coinbase maturity = 100, which is the normal value
3. Generate a number of blocks exactly equal to maturity
4. Run `getbalance` on Core regtest - gives a zero result (this is backed up by examples that always use `generate 101` to give spendable coins)
5. Generate one more block (i.e. the equivalent of `generate 101` overall)
6. `getbalance` now returns 50

Reply:
If that's the case I think we are different than everyone else - This test in NBitcoin passes on bitcoin core and a bunch of other coins. 
https://github.com/MetacoSA/NBitcoin/blob/master/NBitcoin.Tests/AltcoinTests.cs#L156-L169

Reply:
Gosh even more confusing the bitcoin core docs say minconf parameter should default to 1 but I just checked the source and it's defaulted to 0. That being said, there still is something else going on as even with changing the default to 0 I still have to change that >= to > to get the test to pass.

Reply:
Yes, I think you are right and that there is a mistake in that method. I am trying to discern whether our original rationale was valid for incrementing the `countFrom` variable by 1.

I double checked, and the actual definition of a confirmation includes the block a transaction is initially mined in. So my example above is not strictly correct. I do not know why core only gets 50 spendable coins at height 101 then, though.

Reply:
I think its true we need to change to `confirmationCount > network.Consensus.CoinbaseMaturity` @bokobza will also check it out now.